client_authentication_jwks_strategy.go

```diff
@@ -71,7 +71,7 @@ func JKWKSFetcherWithDefaultTTL(ttl time.Duration) func(*DefaultJWKSFetcherStrat
 }
 
 // JWKSFetcherWithCache sets the cache to use.
-func JWKSFetcherWithCache(cache *ristretto.Cache) func(*DefaultJWKSFetcherStrategy) {
+func JWKSFetcherWithCache(cache *ristretto.Cache[string, *jose.JSONWebKeySet]) func(*DefaultJWKSFetcherStrategy) {
```

Comment:
I guess this is technically a breaking change since it will require changes on the user side of things. This is evident by the failing [`oidc-conformity`](https://github.com/ory/fosite/actions/runs/11367049677/job/31618831566?pr=830) test:

```
/go/pkg/mod/github.com/ory/x@v0.0.647/configx/schema_cache.go:16:26: cannot use generic type ristretto.Config[K z.Key, V any] without instantiation
/go/pkg/mod/github.com/ory/x@v0.0.647/configx/schema_path_cache.go:17:30: cannot use generic type ristretto.Config[K z.Key, V any] without instantiation
```

We could get around that by introducing an interface that implements the necessary methods (`Get()`, `SetWithTTL()` and `Wait()`). A downside is that now we cement what the cache can do without making a breaking change (e.g. introducing something like `Del()` would then be breaking).

So I guess it's tradeoff between making a breaking change now, or potentially making a breaking change in the future.

Reply:
On the other hand, the usage of this function seems extremely limited. Not a single use in any public repository on GitHub. At least not with my search query:

https://github.com/search?q=JWKSFetcherWithCache+NOT+is%3Afork&type=code